templates/pioneer-study/esper-study-telemetry.1.parquetmr.txt

```diff
@@ -0,0 +1,42 @@
+message esper-study-telemetry {
+  required group metadata {
+    required int64  Timestamp;
+    required binary documentId (UTF8);
+    required binary pioneerId (UTF8);
+    required binary studyName (UTF8);
+    optional binary geoCity (UTF8);
+    optional binary geoCountry (UTF8);
+    optional binary normalizedChannel (UTF8);
+  }
+  required binary defaultSearchEngine (UTF8);
+  required binary locale (UTF8);
+  required binary os (UTF8);
+  required binary normalizedChannel (UTF8);
+  required binary profileCreationDate (UTF8);
```

Comment:
Even if it's unexpected, some profiles might be missing this. Depending on how important is this field for you, I would encourage making this field `optional`: this way the whole record won't be discarded if this field is missing from the ping.

Reply:
The add-on will send a string "null" in the payload if the value is unavailable, so the payload will always contain the field.